### PR TITLE
New generators 🔥 

### PR DESCRIPTION
**Pull request by TheAwiteb**
# add two new generators 🔥
math and multicolor use [multicolorcaptcha](https://github.com/J-Rios/multicolorcaptcha) library and update code to [black style](https://github.com/psf/black)

## multicolor
Example2

## math
Example2

### CODE REVIEW

**Review by coder2020official — CHANGES_REQUESTED** [review]
What about renaming math generator to numeric?
And, another suggestion: Instead of passing string to captcha options to choose generator, it would be better to create an enum class with all generator names in there.

```
class Generators(str, enum.Enum):
    default: str='default'
    numeric: str='numeric'
    # and so on...

and pass Generators.name to value for generator

```

**Comment by coder2020official** [comment]
Also, just to remind that black style is beta product, we need more testing
https://github.com/psf/black#note-this-is-a-beta-product

**Comment by TheAwiteb** [comment]
> What about renaming math generator to numeric? And, another suggestion: Instead of passing string to captcha options to choose generator, it would be better to create an enum class with all generator names in there.
> 
> ```
> class Generators(str, enum.Enum):
>     default: str='default'
>     numeric: str='numeric'
>     # and so on...
> 
> and pass Generators.name to value for generator
> ```

math generator generates mathematical captcha not numeric why rename it to numeric

**Comment by coder2020official** [comment]
Okay, let it stay math

**Review by TheAwiteb — COMMENTED** [review]
Well I will put the generators in the Enum

**Review by SwissCorePy — CHANGES_REQUESTED** [review]
Looks good to me. Thank you very much!

I also like the idea of putting the generators in an enum class, maybe called `Generator`.

**Comment by coder2020official** [comment]
This will lead to breaking change. Generator should accept string as well.

**Review by coder2020official — COMMENTED** [review]
Just put Generator.name in examples. Captcha should not check whether a class variable is passed, it should still check string's generator name.

**Comment by coder2020official** [comment]
I had to manually edit requirements.txt to resolve conflict. sorry for that.

File: pyTelegramBotCaptcha/telebot_captcha.py [hunk 1]
```diff
@@ -98,82 +127,105 @@ def text(self, your_text: str):
...
+            raise NotImplementedError("This language is not suported yet")
+        if not MIN_TIMEOUT <= timeout <= MAX_TIMEOUT:
+            raise ValueError("timeout must be between 30 and 600 (in seconds)")
+        if not MIN_CODE_LENGTH <= code_length <= MAX_CODE_LENGTH:
+            raise ValueError("code_lenght must be between 4 and 12")
+        if max_attempts < 1:
+            raise ValueError("max_attempts must be at least 1")
+
+        # no need to validate generator because EnumMeta will raise "ValueError: unknown is not a valid Generator"
+        self._generator: str = Generator(generator.lower()).name
```
Comment (TheAwiteb) [inline pyTelegramBotCaptcha/telebot_captcha.py:228]: Hi @coder2020official . I understood from you that the commit to which Enum was added, does not accept strings as generator name, but I did not make any change in the structure of the code. It accepts strings as before, and the function of Enum is only verification as you see here

**Comment by coder2020official** [comment]
I will test as soon as i have time.
I am not sure on your 'black code style' - there are too many changes involving extra spaces. I think it would be good to return to old style, or review will take time.

**Comment by TheAwiteb** [comment]
> I will test as soon as i have time. I am not sure on your 'black code style' - there are too many changes involving extra spaces. I think it would be good to return to old style, or review will take time.

The Black Style was in [one commit](https://github.com/SwissCorePy/pyTelegramBotCAPTCHA/pull/23/commits/b8a75441697425fd375b2b99c780793974d3d6c1) that will make it easier for you to review it. thank you! 🤩

**Comment by TheAwiteb** [comment]
> What about renaming math generator to numeric? And, another suggestion: Instead of passing string to captcha options to choose generator, it would be better to create an enum class with all generator names in there.
> 
> ```
> class Generators(str, enum.Enum):
>     default: str='default'
>     numeric: str='numeric'
>     # and so on...
> 
> and pass Generators.name to value for generator
> ```

The enum has been added. You can close the request

**Review by coder2020official — APPROVED** [review]

**Comment by coder2020official** [comment]
I will test generators. I will write feedback as i test them.

**Review by SwissCorePy — COMMENTED** [review]
Looks great 👍 
But I currently no time to do the tests.

**Review by SwissCorePy — APPROVED** [review]

**Pull request merged**